Dockerfiles: update dockerfiles for automatic builds in quay.io

aanm opened the pull request:
This will allow to use quay.io to build the this docker image without requiring a developer to build it and upload it manually.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
packaging/docker: update docker runtime to 17.10

Signed-off-by: André Martins <andre@cilium.io>

borkmann commented on contrib/packaging/docker/Dockerfile.runtime:
```diff
@@ -22,7 +22,7 @@ RUN \
 apt-get install -y --no-install-recommends make git curl ca-certificates xz-utils \
 # Additional iproute2 build dependencies
-  gcc git pkg-config bison flex && \
+  gcc git pkg-config bison flex build-essential && \
```
Btw, build-essential includes gcc.

jrajahalme commented on Makefile:
```diff
@@ -130,7 +130,6 @@ docker-image-runtime:
 docker-image-builder:
-	cp contrib/packaging/docker/Dockerfile.builder envoy/.
 	cd envoy && docker build -t "cilium/cilium-builder:$(UTC_DATE)" -f Dockerfile.builder .
 	rm envoy/Dockerfile.builder
```
Can't delete the file if it was moved here.

aanm committed:
Dockerfile: point dockerfile to quay.io base images

Since quay.io does not detect symlinks we have to move the Dockefile
directly to the envoy directory.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
envoy: move Dockerfile.builder to envoy directory

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
Makefile: remove docker-image push instructions

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
docs: fix titles formatting

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
docs: add quay.io tutorial

- Add tutorial to update cilium-runtime and cilium-builder images

Signed-off-by: André Martins <andre@cilium.io>

ianvernon commented on contrib/packaging/docker/Dockerfile.runtime:
```diff
@@ -44,7 +44,7 @@ strip /usr/local/clang+llvm/bin/* && \
 # iproute2
 #
-git clone -b v4.10.0 git://git.kernel.org/pub/scm/linux/kernel/git/shemminger/iproute2.git && \
+git clone -b v4.12.0 git://git.kernel.org/pub/scm/network/iproute2/iproute2.git && \
```
Any concern with us updating as crucial a dependency as iproute2 ? I'm wondering if we should do additional testing with these new containers to ensure we won't get bitten by iproute2 bugs. My worry could be unwarranted, though.

aanm replied:
@borkmann said it was ok, also I was not able to build the image any longer for 16.04 and this can be for `1.1`

ianvernon replied:
Great, thanks!

ianvernon commented on Documentation/contributing.rst:
```diff
@@ -751,6 +751,116 @@ for debugging what is going on inside them, for example:
+    :align: center
+    :height: 300px
+
+9. Once the build has started you can disable the Build trigger by clicking in
```
in --> on

ianvernon commented on Documentation/contributing.rst:
```diff
@@ -751,6 +751,116 @@ for debugging what is going on inside them, for example:
+
+9. Once the build has started you can disable the Build trigger by clicking in
+   the wheel.
+10. And click in "Disable Trigger".
```
in --> on

ianvernon commented on Documentation/contributing.rst:
```diff
@@ -751,6 +751,116 @@ for debugging what is going on inside them, for example:
+    :align: center
+    :height: 300px
+
+8. After selecting your branch click in "Start Build".
```
in --> on

ianvernon commented on Documentation/contributing.rst:
```diff
@@ -751,6 +751,116 @@ for debugging what is going on inside them, for example:
+    :align: center
+    :height: 300px
+
+12. Once the build is finish click under Tags (on the left menu).
```
is finished

ianvernon commented on Documentation/contributing.rst:
```diff
@@ -751,6 +751,116 @@ for debugging what is going on inside them, for example:
+    :height: 300px
+
+12. Once the build is finish click under Tags (on the left menu).
+13. Click in the wheel and;
```
in --> on

ianvernon commented on Documentation/contributing.rst:
```diff
@@ -751,6 +751,116 @@ for debugging what is going on inside them, for example:
+    :align: center
+    :height: 300px
+
+15. Write the name of the tag that you want to give for the new image built.
```
new image built --> newly built image

ianvernon commented on Documentation/contributing.rst:
```diff
@@ -751,6 +751,116 @@ for debugging what is going on inside them, for example:
+    :height: 300px
+
+15. Write the name of the tag that you want to give for the new image built.
+16. Confirm the name is correct and click in "Create Tag".
```
in --> on

rlenglet reviewed (approved):
Great docs, thanks!
